Index: README.md
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/README.md b/README.md
--- a/README.md	(revision f891f8525b4865a49091d2672a45dc8e4e9ecabe)
+++ b/README.md	(revision 0d008054c196946491ae174106524a83874731c6)
@@ -4,26 +4,18 @@
 
 ## **Script Hub**
 
-Script Hub is a tool to merge all the scripts into one place.
+ScriptHub is a tool to configure and interact easily with the services and its environments.
 
 <p style="text-align:right;">(<a href="#readme-top">back to top</a>)</p>
 
 ## **Introduction**
 
-This is a hub for scripts, in order to have a place to easily interact with all the scripts,
-instead of having them all scattered and having the way to interact with them all in a different way.
+In this ScriptHub, will have a way to interact with the runners instead of them being scattered.
+
+the issue that we want to solve is to instead of having them all scattered and having the way to interact with them all in a different way.
 This offers a place to unify the interaction with the scripts, making it easier to use.
 
-<p style="text-align:right;">(<a href="#readme-top">back to top</a>)</p>
-
-## **Installation**
-
-To install the Script Hub, follow these steps:
-
-1. Select the folder where you have the project(s)
-2. Clone the repository: **`git clone https://github.com/dariojorge/ScriptsHub.git`**
-3. Make sure you have the Node.js installed in you machine
-4. In case you don't have the Node.js installed go to the page: **`https://nodejs.org/en/download`**
+To start with this tool check the [INTRODUCTION](documentation/INTRODUCTION.md) for the information of how to setup the project
 
 <p style="text-align:right;">(<a href="#readme-top">back to top</a>)</p>
 
@@ -32,8 +24,7 @@
 To use Script Hub, follow these steps:
 
 1. Open a terminal inside the folder scriptsHub
-2. If you are using linux based OS then don't forget to make the runnable using the command **`chmod +x ./scriptHub.sh`
-   **
+2. If you are using linux based OS then don't forget to make the runnable using the command **`chmod +x ./scriptHub.sh`**
 3. To run the scripts check the [Table Of Commands](#table-of-commands) as we are listing there all the current commands
    possible with the script
 4. For more info about the arguments used in the commands check the [Table of Arguments](#table-of-arguments) to know
Index: documentation/CHANGELOG.md
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/documentation/CHANGELOG.md b/documentation/CHANGELOG.md
--- a/documentation/CHANGELOG.md	(revision f891f8525b4865a49091d2672a45dc8e4e9ecabe)
+++ b/documentation/CHANGELOG.md	(revision 0d008054c196946491ae174106524a83874731c6)
@@ -1,5 +1,25 @@
 # Changelog
 
+## v0.1.9
+### Added or Changed
+- Bugfix: Fixing the error when project does not have the folder runners
+- Bugfix: Changing the updateFiles to recreateFiles
+- Bugfix: Changing the default of the recreateFiles to be always true
+
+## v0.1.8
+### Added or Changed
+- Feature: Improving the README by introducing the INTRODUCTION
+- Feature: Changing the createFiles to updateFiles and making it not mandatory by default is false
+
+## v0.1.7
+### Added or Changed
+- Bugfix: Fixing the create files on multiple projects that was failing
+- Bugfix: Fixing if the project is missing inside the runner project
+
+## v0.1.6
+### Added or Changed
+- Feature: Changed the projects envs.json to clean the environments variables
+
 ## v0.1.5
 ### Added or Changed
 - Bugfix: Removing console logs
@@ -49,4 +69,4 @@
     - docker
     - git
     - template
-  - Creation of an example for the runners
+  - Creation of an example for the runners
\ No newline at end of file
Index: documentation/INTRODUCTION.md
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/documentation/INTRODUCTION.md b/documentation/INTRODUCTION.md
new file mode 100644
--- /dev/null	(revision 0d008054c196946491ae174106524a83874731c6)
+++ b/documentation/INTRODUCTION.md	(revision 0d008054c196946491ae174106524a83874731c6)
@@ -0,0 +1,24 @@
+# **RUNNERS EXAMPLE**
+
+## **Introduction**
+This is to show how to quickly setup and use the runners with some examples.
+
+## **Configure NODE**
+Check if the nodejs is installed by running the `node --version`.
+
+You should see as the result something similar to: `v22.16.0`.
+
+In case you don't have the Node.js installed go to the page: **`https://nodejs.org/en/download`**
+
+## **Run the configuration**
+To install the runners project, follow these steps:
+
+1. Select the folder where you have the project(s).
+2. Clone the repository: **`git clone https://github.com/dariojorge/ScriptsHub.git`**
+3. Inside the project ScriptHub run the command: `npm install`
+
+To check if the project is well configured run these two commands:
+```sh
+   ./scriptHub.sh scriptType=runners type=create recreateFiles=<true/false>
+   ./scriptHub.sh scriptType=runners type=update env=<test/int/...>
+```
\ No newline at end of file
Index: documentation/QUARKUS_DEMO.md
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/documentation/QUARKUS_DEMO.md b/documentation/QUARKUS_DEMO.md
--- a/documentation/QUARKUS_DEMO.md	(revision f891f8525b4865a49091d2672a45dc8e4e9ecabe)
+++ b/documentation/QUARKUS_DEMO.md	(revision 0d008054c196946491ae174106524a83874731c6)
@@ -19,7 +19,7 @@
 ```
 
 5. If you now run the command for the runners to create the files, this will create the runners for the project.
-6. Example of the command to run: `./scriptHub.sh scriptType=runners type=create projects=demo-quarkus createFiles=true`
+6. Example of the command to run: `./scriptHub.sh scriptType=runners type=create projects=demo-quarkus recreateFiles=true`
 7. The expected behaviour will be to have the new runner files in the project:
 
 ```bash
Index: package.json
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/package.json b/package.json
--- a/package.json	(revision f891f8525b4865a49091d2672a45dc8e4e9ecabe)
+++ b/package.json	(revision 0d008054c196946491ae174106524a83874731c6)
@@ -1,6 +1,6 @@
 {
   "name": "script-hub",
-  "version": "v0.1.5",
+  "version": "v0.1.9",
   "workspaces": [],
   "dependencies": {}
 }
\ No newline at end of file
Index: scripts/runners/operations/create.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/scripts/runners/operations/create.js b/scripts/runners/operations/create.js
--- a/scripts/runners/operations/create.js	(revision f891f8525b4865a49091d2672a45dc8e4e9ecabe)
+++ b/scripts/runners/operations/create.js	(revision 0d008054c196946491ae174106524a83874731c6)
@@ -1,23 +1,31 @@
 const fs = require("fs");
-const { validateDestinationPath, validateDestinationPathNoCreation, getFiles, getArgValue, convertStringToBoolean, log, error } = require("../../utils/utils");
-const createFilesLabel = "createFiles";
-let runnersPathFrom = "projects/{project}/runners";
-let projectPath = "../{project}";
+const { validateDestinationPath, validateDestinationPathNoCreation, getFiles, getArgValue, convertStringToBoolean, log, error, warning } = require("../../utils/utils");
+const recreateFilesLabel = "recreateFiles";
+let projectInternalPath = "projects/{project}";
+let runnersLabel = `runners`;
+let projectExternalPath = "../{project}";
 let ideaPath = "/.idea";
-let runnersPath = "/runConfigurations";
+const runConfigurationsPath = "/runConfigurations";
 
 const execute = (args) => {
     log('Starting process for the project: ' + args.project);
     const argsObj = buildArgsObj(args);
 
-    const modifiedProjectPath = projectPath.replace("{project}", argsObj.project);
+    const projectsPath = projectInternalPath.replace("{project}", argsObj.project);
+    const modifiedProjectPath = projectExternalPath.replace("{project}", argsObj.project);
+
+    if (!validateDestinationPathNoCreation(projectsPath)) {
+        error(`Project ${argsObj.project} does not exist in the folder projects.`);
+        return;
+    }
+
     if (!validateDestinationPathNoCreation(modifiedProjectPath)) {
         error(`Project ${argsObj.project} does not exist`);
         return;
     }
 
-    const modifiedRunnersPathFrom = runnersPathFrom.replace("{project}", argsObj.project);
-    runnersPath = modifiedProjectPath + ideaPath + runnersPath;
+    const modifiedRunnersPathFrom = `${projectsPath}/${runnersLabel}`;
+    const runnersPath = modifiedProjectPath + ideaPath + runConfigurationsPath;
     validateDestinationPath(modifiedProjectPath + ideaPath);
     validateDestinationPath(runnersPath);
 
@@ -26,21 +34,27 @@
         fromFolderPath: modifiedRunnersPathFrom,
         toListOfFiles: getFiles(runnersPath),
         toFolderPath: runnersPath,
-        createFiles: argsObj.createFiles
+        recreateFiles: argsObj.recreateFiles || true
     }
     copyFiles(fileData);
 };
 
 const buildArgsObj = (argsObj) => {
     return {
-        createFiles: convertStringToBoolean(getArgValue(argsObj.args, createFilesLabel)),
+        recreateFiles: convertStringToBoolean(getArgValue(argsObj.args, recreateFilesLabel)),
         project: argsObj.project
     };
 }
 
 const copyFiles = (fileData) => {
     let fileList = fileData.fromListOfFiles;
-    if (!fileData.createFiles) {
+
+    if(fileList === undefined) {
+        warning(`No runner folder present.`);
+        return;
+    }
+
+    if (!fileData.recreateFiles) {
         fileData.toListOfFiles.forEach(toFile => {
             toFile = toFile.replace(".run", "");
             const index = fileList.indexOf(toFile);
@@ -61,4 +75,4 @@
     log(`Files created where: ${valueCount}`);
 }
 
-module.exports.execute = execute;
+module.exports.execute = execute;
\ No newline at end of file
Index: scripts/runners/operations/settings.json
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/scripts/runners/operations/settings.json b/scripts/runners/operations/settings.json
--- a/scripts/runners/operations/settings.json	(revision f891f8525b4865a49091d2672a45dc8e4e9ecabe)
+++ b/scripts/runners/operations/settings.json	(revision 0d008054c196946491ae174106524a83874731c6)
@@ -29,7 +29,7 @@
           "type": "projectList"
         },
         {
-          "name": "createFiles",
+          "name": "recreateFiles",
           "type": "boolean"
         }
       ]
Index: scripts/runners/operations/utils/additionalCmd.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/scripts/runners/operations/utils/additionalCmd.js b/scripts/runners/operations/utils/additionalCmd.js
--- a/scripts/runners/operations/utils/additionalCmd.js	(revision f891f8525b4865a49091d2672a45dc8e4e9ecabe)
+++ b/scripts/runners/operations/utils/additionalCmd.js	(revision 0d008054c196946491ae174106524a83874731c6)
@@ -1,5 +1,7 @@
 const { execSync } = require("child_process");
 const { convertStringToBoolean, getArgValue, log, error } = require("../../../utils/utils");
+const PROJECT_LABEL = "projectLabel";
+const projectsLabel = "projects";
 const TYPE = {
     EXECUTE: 'EXECUTE',
     ARG: 'ARG'
@@ -11,8 +13,8 @@
 const execute = (args) => {
     const argsObj = buildArgsObj(args);
 
-    argsObj.additionalCmd.forEach(cmd => {
-        processCmd(cmd, argsObj);
+    argsObj.additionalCmd.forEach(cmdObj => {
+        processCmd(cmdObj, argsObj);
     });
 };
 
@@ -20,26 +22,30 @@
     return {
         additionalCmd: argsObj.envData.additionalCmd,
         envData: argsObj.envData,
-        args: argsObj.args
+        args: argsObj.args,
+        project: getArgValue(argsObj.args, projectsLabel)
     };
 }
 
-const processCmd = (cmd, argsObj) => {
-    switch (cmd.type.toUpperCase()) {
+const processCmd = (cmdObj, argsObj) => {
+    switch (cmdObj.type.toUpperCase()) {
         case TYPE.EXECUTE:
-            const conditionBoolean = convertStringToBoolean(cmd.value);
-            if (conditionBoolean) {
-                log(execSync(cmd.cmd, options));
-            }
+            const conditionBoolean = convertStringToBoolean(cmdObj.value);
+            executeCmd(conditionBoolean, argsObj, cmdObj)
             break;
         case TYPE.ARG:
-            const conditionArg = convertStringToBoolean(getArgValue(argsObj.args, cmd.value))
-            if (conditionArg) {
-                log(execSync(cmd.cmd, options));
-            }
+            const conditionArg = convertStringToBoolean(getArgValue(argsObj.args, cmdObj.value))
+            executeCmd(conditionArg, argsObj, cmdObj)
             break;
         default:
-            error(`Cmd type: ${cmd.type} is not configured.`);
+            error(`Cmd type: ${cmdObj.type} is not configured.`);
+    }
+}
+
+const executeCmd = (condition, argsObj, cmdObj) => {
+    if (condition) {
+        const cmd = cmdObj.cmd.replace(PROJECT_LABEL, argsObj.project)
+        log(execSync(cmd, options));
     }
 }
 
Index: scripts/trace/operations/otel-collector-config.yaml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/scripts/trace/operations/otel-collector-config.yaml b/scripts/trace/operations/otel-collector-config.yaml
--- a/scripts/trace/operations/otel-collector-config.yaml	(revision f891f8525b4865a49091d2672a45dc8e4e9ecabe)
+++ b/scripts/trace/operations/otel-collector-config.yaml	(revision 0d008054c196946491ae174106524a83874731c6)
@@ -18,6 +18,13 @@
         match_type: regexp
         span_names:
           - ^.* Client Request Data$
+  filter/exclude_options:
+    spans:
+      exclude:
+        match_type: regexp
+        attributes:
+          - key: "http.request.method"
+            value: "OPTIONS"
 # exporters configure how to send processed data to one or more backends.
 exporters:
   debug:
@@ -37,5 +44,5 @@
   pipelines:
     traces/jaeger:
       receivers: [ otlp ]
-      processors: [ batch/jaeger, filter/filter_client_request_data_spans ]
+      processors: [ batch/jaeger, filter/filter_client_request_data_spans, filter/exclude_options ]
       exporters: [ debug, otlp/jaeger ]
\ No newline at end of file
Index: scripts/utils/utils.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/scripts/utils/utils.js b/scripts/utils/utils.js
--- a/scripts/utils/utils.js	(revision f891f8525b4865a49091d2672a45dc8e4e9ecabe)
+++ b/scripts/utils/utils.js	(revision 0d008054c196946491ae174106524a83874731c6)
@@ -2,7 +2,7 @@
 
 // LOGS
 const log = (message) => console.log(message);
-const warning = (message) => console.warning(message);
+const warning = (message) => console.warn(message);
 const error = (message) => console.error(message);
 module.exports.log = log;
 module.exports.warning = warning;
